src/controllers/userController.js

```diff
@@ -157,8 +161,7 @@ userController.put('/editUserRole/:id', async (req, res) => {
 });
 
 userController.get('/getMyEvents', async (req, res) => {
-    const userId = req.requester._id;
-
+    const userId = req.requester?._id;
```

Comment:
why not in the try catch?

Reply:
When is in try-catch the error is:
"error": "Cannot read properties of undefined (reading '_id')";
when using "?"operator the error is:
"error":"User not found"
Not sure which one is more useful for FE?

Reply:
I didn't mention the question mark, keep it, but move the code inside the try catch, as you did in the code above